modules/search/instant-search/components/search-result-product.jsx

```diff
@@ -0,0 +1,64 @@
+/** @jsx h */
+
+/**
+ * External dependencies
+ */
+import { h, Component } from 'preact';
+
+/**
+ * Internal dependencies
+ */
+import SearchResultComments from './search-result-comments';
+import PhotonImage from './photon-image';
+
+class SearchResultProduct extends Component {
+	render() {
+		const { result_type, fields, highlight } = this.props.result;
+		if ( result_type !== 'post' ) {
+			return null;
+		}
+
+		const firstImage = Array.isArray( fields[ 'image.url.raw' ] )
+			? fields[ 'image.url.raw' ][ 0 ]
+			: fields[ 'image.url.raw' ];
+
+		return (
+			<li className="jetpack-instant-search__search-result-product">
+				<h3>
+					<a
+						className="jetpack-instant-search__result-product-title"
+						href={ `//${ fields[ 'permalink.url.raw' ] }` }
+						onClick={ this.props.onClick }
+						rel="noopener noreferrer"
+						target="_blank"
+						//eslint-disable-next-line react/no-danger
+						dangerouslySetInnerHTML={ { __html: highlight.title } }
+					/>
+				</h3>
+				{ firstImage && (
+					<PhotonImage
+						className="jetpack-instant-search__search-result-product-img"
+						src={ `//${ firstImage }` }
+						alt=""
+					/>
+				) }
+				<div
+					className="jetpack-instant-search__search-result-product-content"
+					//eslint-disable-next-line react/no-danger
+					dangerouslySetInnerHTML={ {
+						__html: highlight.content.join( ' ... ' ),
```

Comment:
same here we need to ensure highlight.content is always properly escaped

Reply:
cc @gibrown

Reply:
ditto. done in endpoint